spec/controllers/districts_controller_spec.rb

```diff
@@ -0,0 +1,100 @@
+require 'rails_helper'
+
+RSpec.describe DistrictsController, type: :controller do
+  let(:valid_session) { {} }
+  let(:district_with_valid_name) do
+    { name: 'Личаківський', city_id: district.city_id }
+  end
+
+  let(:district_with_invalid_name) do
+    { name: '' }
+  end
+
+  let(:user_admin) { FactoryBot.create(:user_admin) }
```

Comment:
You should test all actions with user without admin access too. And check that he do not have access